File: ios/coffee-oc/coffee/coffee/DataModels/Group.m
//
//  Group.m
//  coffee
//
//  Created by achiel on 14-10-6.
//  Copyright (c) 2014年 1clip. All rights reserved.
//

#import "Group.h"

@implementation Group

-(NSMutableArray<Friend, Optional>*) friends {
    if (!_friends) {
        _friends = (NSMutableArray<Friend, Optional>*)[NSMutableArray array];
    }
    
    return _friends;
}

@end

@implementation Friend

-(NSMutableArray<Group, Optional>*) groups {
    if (!_groups) {
        _groups = (NSMutableArray<Group, Optional>*)[NSMutableArray array];
    }
    
    return _groups;
}

@end

File: ios/coffee/coffee/Data Models/Group.swift
//
//  Group.swift
//  coffee
//
//  Created by achiel on 14-10-4.
//  Copyright (c) 2014年 1Clip. All rights reserved.
//

import Foundation

public class Group : IdObject {
    
    public var name: NSString!
    
    // This is only for internal usage
    internal var friends: Array<Friend>!
}

public class Friend : Character {
    
    public var groups: Array<Group>!
}